Program.cs

```diff
@@ -3,16 +3,91 @@
 // Copyright (c) Metamation India.                                                
 // ---------------------------------------------------------------------------------------
 // Program.cs                                                                     
-// 
-// 
-// 
+// Program to implement double.Parse method that takes a string and returns a double.
 // ---------------------------------------------------------------------------------------
+
 namespace Training {
    #region class Program --------------------------------------------------------------------------
-   internal class Program {
+   /// <summary>Double.Parse</summary>
+   public class DoubleParse {
       #region Method ------------------------------------------------
-      static void Main (string[] args) {
-         Console.WriteLine ("Hello, World!");
+      /// <summary>Gets the user input and prints the desired output</summary>
+      static void Main () {
+         Console.Write ("Enter a value to be converted into double: ");
+         string input = Console.ReadLine ().ToLower ();
+         Console.WriteLine (IsValid (input) ? Parse (input) : "Invalid Input");
+      }
+
+      /// <summary>Checks if the user input is valid</summary>
+      /// <param name="s">user input</param>
+      /// <returns>Returns true if the input is valid</returns>
+      public static bool IsValid (string s) {
+         if (s.Any (x => x != '+' && x != '-'                 // Checks if all are valid characters.
+                      && x != '.' && x != 'e'
+                      && !char.IsDigit (x))) return false;
+         if (s.All (x => !char.IsDigit (x))) return false;    // Checks if atleast one digit is present.

```

Comment:
When this condition will be hit? Can you give me an example?

Reply:
Was written for inputs like "+e." where all characters are valid but none of them is a digit. 
But removed this condition and added extra invalid conditions such that it also covers inputs including these and other necessary ones.